src/main/java/de/tum/in/www1/artemis/service/connectors/vcs/AbstractVersionControlService.java

```diff
@@ -163,6 +164,9 @@ public String getOrRetrieveBranchOfStudentParticipation(ProgrammingExerciseStude
     @Override
     public String getOrRetrieveBranchOfExercise(ProgrammingExercise programmingExercise) {
         if (programmingExercise.getBranch() == null) {
```

Comment:
To reduce the nesting, we could invert this conditional return `branch`in the section below.

Reply:
I see your point